File: src/Entity/Livraison.php
<?php

namespace App\Entity;

use App\Repository\LivraisonRepository;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: LivraisonRepository::class)]
class Livraison
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\OneToOne(inversedBy: 'Livraison', cascade: ['persist', 'remove'])]
    #[ORM\JoinColumn(nullable: false)]
    private ?Commande $commande = null;

    #[ORM\Column(length: 50)]
    private ?string $statut = null;

    #[ORM\Column(length: 100)]
    private ?string $transporteur = null;

    #[ORM\Column(length: 20)]
    private ?string $numTelTransporteur = null;

    #[ORM\Column(type: Types::DATETIME_MUTABLE, nullable: true)]
    private ?\DateTimeInterface $dateLivraison = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function setId(int $id): static
    {
        $this->id = $id;

        return $this;
    }

    public function getCommande(): ?Commande
    {
        return $this->commande;
    }

    public function setCommande(Commande $commande): static
    {
        $this->commande = $commande;

        return $this;
    }

    public function getStatut(): ?string
    {
        return $this->statut;
    }

    public function setStatut(string $statut): static
    {
        $this->statut = $statut;

        return $this;
    }

    public function getTransporteur(): ?string
    {
        return $this->transporteur;
    }

    public function setTransporteur(string $transporteur): static
    {
        $this->transporteur = $transporteur;

        return $this;
    }

    public function getNumTelTransporteur(): ?string
    {
        return $this->numTelTransporteur;
    }

    public function setNumTelTransporteur(string $numTelTransporteur): static
    {
        $this->numTelTransporteur = $numTelTransporteur;

        return $this;
    }

    public function getDateLivraison(): ?\DateTimeInterface
    {
        return $this->dateLivraison;
    }

    public function setDateLivraison(?\DateTimeInterface $dateLivraison): static
    {
        $this->dateLivraison = $dateLivraison;

        return $this;
    }
}

File: src/main/java/tn/esprit/entities/Livraison.java
package tn.esprit.entities;


import jakarta.persistence.*;
import javax.validation.constraints.*;   // ✅ This is what YOU should use (based on your dependencies)
import java.time.LocalDateTime; // ✅ required for updatedAt
import java.util.ArrayList;
import java.util.List;

@Entity
public class Livraison {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @OneToOne(optional = false)
    @JoinColumn(name = "commande_id")
    @NotNull(message = "La commande associée est obligatoire")
    private Commande commande;

    @Column(length = 50, nullable = false)
    @NotBlank(message = "Le statut de la livraison est obligatoire")
    @Pattern(
            regexp = "En Cours|Livrée|Annulée",
            message = "Statut de livraison invalide"
    )
    private String statut;

    @Column(length = 100, nullable = false)
    @NotBlank(message = "Le transporteur est obligatoire")
    @Size(
            min = 3,
            max = 100,
            message = "Le nom du transporteur doit contenir entre 3 et 100 caractères"
    )
    private String transporteur;

    @Column(length = 20, nullable = false)
    @NotBlank(message = "Le numéro de téléphone du transporteur est obligatoire")
    @Pattern(
            regexp = "^\\+?[0-9\\s\\-]+$",
            message = "Le numéro de téléphone est invalide"
    )
    private String numTelTransporteur;

    @Column
    private LocalDateTime dateLivraison;

    // Getters and Setters

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public Commande getCommande() {
        return commande;
    }

    public void setCommande(Commande commande) {
        this.commande = commande;
    }

    public String getStatut() {
        return statut;
    }

    public void setStatut(String statut) {
        this.statut = statut;
    }

    public String getTransporteur() {
        return transporteur;
    }

    public void setTransporteur(String transporteur) {
        this.transporteur = transporteur;
    }

    public String getNumTelTransporteur() {
        return numTelTransporteur;
    }

    public void setNumTelTransporteur(String numTelTransporteur) {
        this.numTelTransporteur = numTelTransporteur;
    }

    public LocalDateTime getDateLivraison() {
        return dateLivraison;
    }

    public void setDateLivraison(LocalDateTime dateLivraison) {
        this.dateLivraison = dateLivraison;
    }
}
